packages/graphiql/src/components/GraphiQL.tsx

```diff
@@ -545,11 +547,15 @@ export function GraphiQLInterface(props: GraphiQLInterfaceProps) {
                             {tab.title}
                           </Tab.Button>
                           <Tab.Close
-                            onClick={() => {
-                              if (editorContext.activeTabIndex === index) {
-                                executionContext.stop();
+                            onClick={async () => {
+                              if (
```

Comment:
love this! however, i think it would make more sense to add this to the context methods in the sdk rather than in graphiql reference implementation itself. so we will add this to `closeTab()` instead, so it can be available to both `graphiql` and @graphiql/react` users and graphiql plugin developers as well
in `editor/context.tsx`, you can update the method to this
```ts
  const closeTab = useCallback<EditorContextType['closeTab']>(
    async index => {
      if (await closeTabConfirmation(index)) {
        setTabState(current => {
          const updated = {
            tabs: current.tabs.filter((_tab, i) => index !== i),
            activeTabIndex: Math.max(current.activeTabIndex - 1, 0),
          };
          storeTabs(updated);
          setEditorValues(updated.tabs[updated.activeTabIndex]);
          onTabChange?.(updated);
          return updated;
        });
      }
    },
    [onTabChange, setEditorValues, storeTabs, closeTabConfirmation],
  );

```